types/validator.go

```diff
@@ -136,12 +136,9 @@ func (v *Validator) ValidateBasic() error {
 }
 
 // ValidatePubKey performs basic validation on the public key.
+// Note that it is not an error for the public key to be nil.
 func (v *Validator) ValidatePubKey() error {
-	if v.PubKey == nil {
-		return errors.New("validator does not have a public key")
-	}
-
-	if len(v.PubKey.Bytes()) != bls12381.PubKeySize {
+	if v.PubKey != nil && len(v.PubKey.Bytes()) != bls12381.PubKeySize {
```

Comment:
Why we combine these two errors?

Reply:
changed the solution, so this comment is outdated